src/conv/invokers/impl_gemm_dynamic.cpp

```diff
@@ -579,7 +579,7 @@ InvokerFactory MakeImplGemmDynamicForwardXdlopsNHWCInvokerFactory(
             trans_output_idx = idx++;
     }
 
-    const size_t cast_offset = is_nchw ? (trans_output_offset + trans_output_size) : 0;
+    const size_t cast_offset = is_nchw ? (((trans_output_offset + trans_output_size + 3) >> 2) << 2) : 0;
```

Comment:
I think we need add some comments here, since this is quite hacky

Reply:
Right. Added.